### DISCUSSION TITLE
How to customize TOC

### DISCUSSION THREAD

**Discussion by BeeGass** [allejo/jekyll-toc#58] [Q&A]
`"The very first step is to find all of the headings in the given HTML"`

When looking through and trying to follow @allejo [guide](https://allejo.io/blog/a-jekyll-toc-without-plugins-or-javascript/) I had trouble understanding this part. Coming from a newbie difficult to this sounds like you went into the `_site` folder and looked at the markdown-ify-ed html and found the corresponding headers. This seems unintuitive and based off what I have looked at, not the case. Im hoping someone might be able to fill the gaps on allejo's blog post.

**Comment by allejo** [comment]
That blog post you're referring to is now outdated as this project is now using a different method of generating the ToC via markdown. However, that blog post is intended to be more of a "how does this Liquid magic even work?!" technical explanation. I can probably clarify that in the post and mark it as an outdated implementation.

On the topic of customizing the ToC, what are you looking to customize?

**Reply by BeeGass** [reply]
I was able to figure it out. However, what I was trying to figure out was how to reference the ToC in specific when in CSS. Now, my solution was to go into the liquid code (i know its not that difficult) and read the commented example given as well as your test cases. What I was proposing was to create a more formal guide/example that shows that the id or class that you reference in CSS in defined by YOU and where to define those id or class names.
